lib/sinon/collection.js

```diff
@@ -46,7 +46,13 @@ var collection = {
     },
 
     resetHistory: function resetHistory() {
-        each(this, "resetHistory");
+        getFakes(this).forEach(function (fake) {
+            var method = fake.resetHistory || fake.reset;
+
+            if (method) {
```

Comment:
Are they fakes that have neither `resetHistory` nor `reset`? It seems like we could simplify this a bit and be in the same place:

```js
(fake.resetHistory || fake.reset).call(fake);
```